docs/content/docs/deep/componets.md

```diff
@@ -0,0 +1,113 @@
+
+When configuring the container, the application specifies which DDSes and DataObjects implementations the container supports, which implicitly selects the concrete implementation (which package version and which copy of the package if there are multiple) the container will use for the DDses and DataObjects.
+
+Smaller libraries/components which the applications uses to work with data within the container should define version ranges of the Fluid client libraries they are compatible with, and the application should ensure all code using the same container is using the same copies of the Fluid client packages.
+Some exceptions to this are known to work: TODO: list working/supported exceptions here.
+
+An application can select which DDses (TODO: or DataObjects?) in the container to load, allowing both reading and writing to containers which are only partially loaded.
+(TODO: note scalability implications due to summary client, op bandwidth).
+
+Within a container, all DDses are synchronized, and operations can between different DDSes are ordered.
+Each DDS has its own merge resolution, so there is no way to do things like multi DDS transactions, or have changes in one DDS impact how a change is merged in another, which can limit the fidelity of operations like moves of content between DDSes, even within a container.
+
+The Fluid Framework provides a garbage collector which can delete DDSes when they become unreferenced by the container (no reachable from the root DDS, which is typically the shared directory created from the container schema).
+
+### Data Objects
+
+Data objects are an under development abstraction on-top of DDSes (see below) which can allow reusable logic build on DDSes to be easily reused.
+
+Currently no public APIs exist for defining new data-object types, but the existing experimental ones can be used in container schema.
+
+### Distributed Data Structures: DDSes
+
+A data structure in a container, with its own self contained editing and merge resolution logic.
+May contain handles to other DDSes.
+
+Some DDSes (mainly SHaredTree) provide their own functionality for further componentization of their content.
+
+While while the user of the container can choose which DDSes to load, DDSes themselves currently can not support being only partially loaded: either the whole DDS is available or not.
+The one exception to this is blobs which the user of the container can asynchronously upload then insert handle to them into the DDS.
+Future work (See SharedTree) may enable some DDSes to load subsets of their data on demand.
+
+## SharedTree
+
+SharedTree is design to encourage componentization by subtree / schema.
+Typical usage is a given component defines its schema and logic which works with that schema.
+This schema/component is then referenced/depended on by its parent, adding its schema as a child.
```

Comment:
Typical software composition relies on encapsulation / abstractions (in many cases - via object-oriented programming), i.e. not exposing guts of a library. I'd state here that such model is not supported, with implication that the only model supported is the one you describe.

Reply:
That said, maybe it's worth the reference to what is meant by "schema" (when I read this document, I think about "schema" as pure data schema), and the fact that SharedTree could have code bundled with data schema. So, to some extent it's OOP, but you have to expose all the guts (declare schema to the world) and no ability to add access specifiers (public / private), I believe (correct me if I do not know about something).